packages/core-sdk/src/resources/royalty.ts

```diff
@@ -387,7 +389,7 @@ export class RoyaltyClient {
       txHash,
       rpcClient: this.rpcClient,
       txOptions,
-    });
+    }) as Promise<TransactionResponse>;
```

Comment:
we need to minimize the amount of casing in the code. Lets make those method updates so we can avoid doing it, thus making the code cleaner.